static/docs/commands-reference/update.md

```diff
@@ -37,3 +37,31 @@ their DVC-files (via fields `lock: true` and `rev_lock`). For a locked stage,
   problems arise, otherwise 1.
 
 - `-v`, `--verbose` - displays detailed tracing information.
+
+## Example
```

Comment:
we should remember to update the description above and mention that `import-url` locks artifacts as well. and that `dvc update` updates both now.

Reply:
also, second Note about dvc repro looks redundant to me.

also, `This is the only command that does so with locked stages, and import stages are locked by default.` - not sure we need this.

it's better to describe what's happening here from the user perspective in the first place and leave internal mechanics to `>` or expandable notes. From a user perspective it's simple - download a new version and update DVC-file. I don't care if it unlocks file, if it runs repro inside, etc - these details should go to notes.

Reply:
Updated! And yes that note was totally redundant. Reorganized the whole description in 77f9533 but let me think about your comment on the user's perspective...

Reply:
OK. I like this idea to focus command references (and docs in general) to user-focused descriptions and explanations. BUT we currently don't really have that in many places. For example:

- The description in https://dvc.org/doc/commands-reference/add goes into detail on what happens "under the hood" after the 1st paragraph.
- https://dvc.org/doc/commands-reference/checkout also has a pretty detailed and long description
- https://dvc.org/doc/commands-reference/commit also has an under-the-hood style bullet list in its description
  etc.

So seems to me that for consistency's sake we'd need to make a major review of all command refs in order to take this new direction?

All that said, I did rewrite this one description focusing on the user's perspective in 54fbeb3. (So the summarized changes are https://github.com/jorgeorpinel/dvc.org/compare/b8d4512...54fbeb3.)

Reply:
I would keep all these details. Let me correct my previous comment. I think description can and should include as much details as possible, but it should be user-centric anyway. We should start them with some high-level description - `dvc commit` is more or less good example. And it's totally fine to include more under-the-hood details. At least when they describe some user-visible things - like a file moved to cache, DVC-file created with a specific field. I would avoid describing specific mechanism certain behavior implemented with. I think this is similar to public vs private methods in software engineering :)

In our specific case. Let's remove this lock/unlock stuff just because we are going to change it soon, locking of these import/import-url stages won't depend on `lock: true` - it will be by definition for those stages.

Reply:
> I would avoid describing specific mechanism certain behavior implemented with. I think this is similar to public vs private methods in software engineering :)

Got it!

> Let's remove this lock/unlock stuff just because we are going to change it soon

OK, removed in debb558